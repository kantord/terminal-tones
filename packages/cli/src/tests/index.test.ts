import { it, expect, describe } from 'vitest';
import { execa } from 'execa';
import stripAnsi from 'strip-ansi';
import { fileURLToPath } from 'node:url';
import path from 'node:path';

const here = path.dirname(fileURLToPath(import.meta.url));
const tsEntry = path.resolve(here, '..', 'index.ts');

describe('terminal-tones --help', () => {
  it('prints help and exits 0', async () => {
    const { stdout, stderr, exitCode } = await execa(
      'node',
      ['--import', 'tsx', tsEntry, '--help'],
      { reject: false }
    );
    expect(exitCode).toBe(0);
    expect(stderr).toBe('');
    expect(stripAnsi(stdout)).toMatchSnapshot();
  });
});
