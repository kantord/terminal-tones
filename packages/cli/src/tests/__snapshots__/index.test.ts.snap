// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`terminal-tones --help > prints help and exits 0 1`] = `
"Usage: terminal-tones [options] [command]

A command line tool for generating terminal color schemes

Options:
  -h, --help         display help for command

Commands:
  from-image <path>  Generate a color scheme from an image
  help [command]     display help for command"
`;
